#! /bin/sh /usr/share/dpatch/dpatch-run
## 31_gnu-kfreebsd.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: GNU/kFreeBSD support (not accepted upstream)

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad cdrtools-2.01.01~/RULES/os-gnu-kfreebsd.def cdrtools-2.01.01/RULES/os-gnu-kfreebsd.def
--- cdrtools-2.01.01~/RULES/os-gnu-kfreebsd.def	1970-01-01 01:00:00.000000000 +0100
+++ cdrtools-2.01.01/RULES/os-gnu-kfreebsd.def	2006-03-19 21:25:02.244014400 +0100
@@ -0,0 +1 @@
+MANSTYLE=	bsd
diff -urNad cdrtools-2.01.01~/cdda2wav/interface.c cdrtools-2.01.01/cdda2wav/interface.c
--- cdrtools-2.01.01~/cdda2wav/interface.c	2006-03-19 21:25:01.666102256 +0100
+++ cdrtools-2.01.01/cdda2wav/interface.c	2006-03-19 21:28:23.977346272 +0100
@@ -491,12 +491,12 @@
 #endif
        break;
 
-#if defined (__linux__) || defined (__FreeBSD__) || defined(__DragonFly__)
+#if defined (__linux__) || defined (__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #if defined (__linux__)
     case SCSI_CDROM_MAJOR:     /* scsi cd */
     default:			/* for example ATAPI cds */
 #else
-#if defined (__FreeBSD__) || defined(__DragonFly__)
+#if defined (__FreeBSD__) || defined(__DragonFly__)  || defined(__FreeBSD_kernel__)
 #if __FreeBSD_version >= 501113
     case 4:	/* GEOM */
 	/* FALLTHROUGH */
diff -urNad cdrtools-2.01.01~/cdda2wav/ioctl.c cdrtools-2.01.01/cdda2wav/ioctl.c
--- cdrtools-2.01.01~/cdda2wav/ioctl.c	2006-03-19 21:24:53.000000000 +0100
+++ cdrtools-2.01.01/cdda2wav/ioctl.c	2006-03-19 21:30:43.685107464 +0100
@@ -63,7 +63,7 @@
 static struct cdrom_read_audio arg;
 #endif
 
-#if (defined(__FreeBSD__) && __FreeBSD_version >= 400014) || defined(__DragonFly__)
+#if (defined(__FreeBSD__) && __FreeBSD_version >= 400014) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 static unsigned sector_size;
 #endif
 
@@ -76,7 +76,7 @@
 	int fAudioMode;
 	unsigned uSectorsize;
 {
-#if	(defined(__FreeBSD__) && __FreeBSD_version >= 400014) || defined(__DragonFly__)
+#if	(defined(__FreeBSD__) && __FreeBSD_version >= 400014) || defined(__DragonFly__)  || defined(__FreeBSD_kernel__)
 	if (scgp && scgp->verbose)
 		fprintf(stderr, "EnableCdda_cooked (CDRIOCSETBLOCKSIZE)...\n");
 
@@ -172,7 +172,7 @@
 	    fprintf( stderr, "can't get TocEntry #%d lba (error %d).\n", i+1, err );
 	    exit( MEDIA_ERROR );
 	}
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 	entry[i].cdte_addr.lba = be32_to_cpu(entry[i].cdte_addr.lba);
 #endif
     }
@@ -184,7 +184,7 @@
 	fprintf( stderr, "can't get TocEntry LEADOUT lba (error %d).\n", err );
 	exit( MEDIA_ERROR );
     }
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
     entry[i].cdte_addr.lba = be32_to_cpu(entry[i].cdte_addr.lba);
 #endif
 
@@ -212,7 +212,7 @@
 {
       /* trash the cache */
 
-#if	defined(__FreeBSD__) || defined(__DragonFly__)
+#if	defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #if	defined(__FreeBSD__) && __FreeBSD_version >= 501112
       pread(global.cooked_fd, (void *) &p[0], 3*CD_FRAMESIZE_RAW,
           find_an_off_sector(lSector, SectorBurstVal)*CD_FRAMESIZE_RAW);
@@ -287,7 +287,7 @@
   static int nothing_read = 1;
 
 /* read 2352 bytes audio data */
-#if	defined(__FreeBSD__) || defined(__DragonFly__)
+#if	defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #if	defined(__FreeBSD__) && __FreeBSD_version >= 501112
     if (x && x->verbose) {
 	fprintf(stderr, "ReadCdRom_cooked (pread)...\n");
@@ -428,7 +428,7 @@
 {
     struct cdrom_subchnl sub_ch;
 
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
     struct cd_sub_channel_info sub_ch_info;
 
     if (x && x->verbose) {
@@ -466,7 +466,7 @@
           return NULL;
       }
       case GET_POSITIONDATA:
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
       sub_ch.data_format = CD_CURRENT_POSITION;
 #endif
 #if defined (__linux__)
diff -urNad cdrtools-2.01.01~/cdda2wav/mycdrom.h cdrtools-2.01.01/cdda2wav/mycdrom.h
--- cdrtools-2.01.01~/cdda2wav/mycdrom.h	2006-03-19 21:24:53.000000000 +0100
+++ cdrtools-2.01.01/cdda2wav/mycdrom.h	2006-03-19 21:47:52.452711016 +0100
@@ -58,7 +58,7 @@
 
 #   endif /* if 0 */
 #  else /* not Sun SVR4 */
-#   if defined __FreeBSD__ || defined __NetBSD__ || defined __OpenBSD__ || defined __DragonFly__
+#   if defined __FreeBSD__ || defined __NetBSD__ || defined __OpenBSD__ || defined __DragonFly__  || defined __OpenBSD__
 #    if (defined(__FreeBSD__) && __FreeBSD_version < 228000) || !defined(CDIOCREADAUDIO)
 #     undef HAVE_IOCTL_INTERFACE
 #    else
diff -urNad cdrtools-2.01.01~/libscg/scsihack.c cdrtools-2.01.01/libscg/scsihack.c
--- cdrtools-2.01.01~/libscg/scsihack.c	2006-03-19 21:24:53.000000000 +0100
+++ cdrtools-2.01.01/libscg/scsihack.c	2006-03-19 21:48:46.949426256 +0100
@@ -128,7 +128,7 @@
 
 #endif	/* linux */
 
-#if	defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#if	defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)  || defined(__FreeBSD_kernel__) 
 #define	SCSI_IMPL		/* We have a SCSI implementation for *BSD */
 
 #include "scsi-bsd.c"
